package/base-files/files/sbin/sysupgrade

```diff
@@ -101,6 +101,26 @@ EOF
 # prevent messages from clobbering the tarball when using stdout
 [ "$CONF_BACKUP" = "-" ] && export VERBOSE=0
 
+wget_if_URL(){
+	local sed_head='^.*\('
+	local url_pattern='\(https\?\|ftp\)://[^ ]*'
+	local sed_tail='\).*$'
+	local url=$(echo "$ARGV" | sed -ne "s#$sed_head$url_pattern$sed_tail#\1#p")
+	[ -z "$url" ] && return 1
+	rm -f /tmp/sysupgrade-URL.bin
+	wget "$url" -O "/tmp/sysupgrade-URL.bin" || {
+		echo "Wget return error $? for url '$url'"
+		echo "Exiting"
+		rm -f /tmp/sysupgrade-URL.bin
+		exit 1
+	}
+	ARGV=$(echo "$ARGV" | sed -e "s#$url_pattern#/tmp/sysupgrade-URL.bin#")
+	return 0
+}
+
+#if URL has been specified then downloading image file via wget
+wget_if_URL
+
```

Comment:
We should implement that in a separate PR, unrelated to the Mikrotik support and probably encapsulate the logic in `get_image()`